akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/LoggingEventFilter.scala

```diff
@@ -1,473 +1,188 @@
 
   /**
-   * Apply this filter while executing the given code block. Care is taken to
-   * remove the filter when the block is finished or aborted.
+   * Matching events with an included `throwable` that is a class or subclass of the given
+   * `Throwable` `ClassTag`.
    */
-  def intercept[T](code: => T)(implicit system: ActorSystem[_]): T = {
-    interceptLogger("")(code)
-  }
+  def withCause[A <: Throwable: ClassTag]: LoggingEventFilter
 
   /**
-   * Apply this filter while executing the given code block. Care is taken to
-   * remove the filter when the block is finished or aborted.
+   * Matching events with MDC containing all entries of the given `Map`.
+   * The event MDC may have more entries than the given `Map`.
+   */
+  def withMdc(newMdc: Map[String, String]): LoggingEventFilter
+
+  /**
+   * Matching events for which the supplied function returns`true`.
    */
-  def interceptLogger[T](loggerName: String)(code: => T)(implicit system: ActorSystem[_]): T = {
-    TestAppender.setupTestAppender(loggerName)
-    TestAppender.addFilter(loggerName, this)
-    // FIXME #26537 other setting for leeway
-    import akka.actor.typed.scaladsl.adapter._
-    val leeway = TestKitExtension(system.toUntyped).TestEventFilterLeeway
-    try {
-      val result = code
-      if (!awaitDone(leeway))
-        if (todo > 0)
-          throw new AssertionError(s"timeout ($leeway) waiting for $todo messages on $this")
-        else
-          throw new AssertionError(s"received ${-todo} excess messages on $this")
-      result
-    } finally {
-      TestAppender.removeFilter(loggerName, this)
-    }
-  }
+  def withCustom(newCustom: Function[LoggingEvent, Boolean]): LoggingEventFilter
 
-  /*
-   * these default values are just there for easier subclassing
+  /**
+   * @return `true` if the event matches the conditions of the filter.
    */
-  protected val source: Option[String] = None
-  protected val message: Either[String, Regex] = Left("")
-  protected val complete: Boolean = false
+  def matches(event: LoggingEvent): Boolean
 
   /**
-   * internal implementation helper, no guaranteed API
+   * Apply this filter while executing the given code block. Care is taken to
+   * remove the filter when the block is finished or aborted.
    */
-  protected def doMatch(src: String, msg: Any) = {
-    val msgstr = if (msg != null) msg.toString else "null"
-    (source.isDefined && source.get == src || source.isEmpty) &&
-    (message match {
-      case Left(s)  => if (complete) msgstr == s else msgstr.startsWith(s)
-      case Right(p) => p.findFirstIn(msgstr).isDefined
-    })
-  }
+  def intercept[T](code: => T)(implicit system: ActorSystem[_]): T
 
+  /**
+   * Apply this filter while executing the given code block. Care is taken to
+   * remove the filter when the block is finished or aborted.
+   */
+  def interceptLogger[T](loggerName: String)(code: => T)(implicit system: ActorSystem[_]): T
```

Comment:
I would have preferred `loggerName` to be (or look like from the users perspective) another filter property for consistency and have just a single `intercept`

Reply:
@johanandren That is obviously much better. Fixed in 75791d2c55257476211dee40641f16f5ca967c9c

(the background was that this loggerName is where the TestAppender is installed, and not an exact condition of the LoggingEvent, but that is an implementation detail)